infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/util/EnvVars.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.util;
+
+import io.fabric8.kubernetes.api.model.EnvVar;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/** Utility class for dealing with environment variables */
+public class EnvVars {
+
+  private static final Pattern REFERENCE_PATTERN = Pattern.compile("\\$\\(\\w+\\)");
```

Comment:
There are also variable notation with curly brackets, e.g. `export MY_VAR=${ANOTHER_VAR}`
As far as I understand this class won't support it. Is it deliberately made decision to skip support of it?

Reply:
It seems this pattern supports `$(some)` which is rather command substitution. See http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_03
Can you elaborate why we don't support more common cases of expansion like `$MY_VAR` or `${MY_VAR}`?

Reply:
And since it is command substitution it should not even work like that for regular shells. So, it is completely custom feature that just looks like what shell do.

Reply:
Should we implement this feature with a custom notation or use something widely known like shell expansion? 
@metlos @sleshchenko @skabashnyuk @l0rd WDYT?

Reply:
This is specifically for variable expansion in kubernetes, which only allows the `$(,,,)` syntax. See https://github.com/kubernetes/kubernetes/blob/master/third_party/forked/golang/expansion/expand.go

Reply:
Or the docs: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#envvar-v1-core

Reply:
Since we assume k8s-based infrastructure, I think we should stick to the conventions available there.

Reply:
Now it is clear. Thanks for the pointer. BTW here is the corresponding user docs https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/#using-environment-variables-inside-of-your-config
Could you add a link to a doc to the source code to make it less confusing to those who don't know that k8s uses special notation?

Reply:
Good idea